### PR TITLE
feat: konflux cluster pool in new QE AWS account

### PR DESCRIPTION
**Pull request by psturc**
https://issues.redhat.com/browse/KONFLUX-4288

* created new cluster pool "konflux-cluster-pool" with the new QE AWS credentials
* updated e2e-tests job (which is optional) to point to this new cluster pool to test it out
* follow-up PR will update all konflux-related jobs as well

### CODE REVIEW

File: ci-operator/config/konflux-ci/e2e-tests/konflux-ci-e2e-tests-main.yaml [hunk 1]
```diff
@@ -42,7 +42,7 @@ tests:
     cloud: aws
     labels:
       region: us-east-2
-    owner: rhtap
+    owner: konflux
```
Comment (psturc) [inline ci-operator/config/konflux-ci/e2e-tests/konflux-ci-e2e-tests-main.yaml:45]: I changed only this (optional) job in order to test that cluster pool works correctly. a follow-up PR will be created for switching to konflux cluster pool in all konflux-related jobs

**Review by kasemAlem — APPROVED** [review]

File: clusters/hosted-mgmt/hive/pools/konflux/konflux-ocp-4-15-0-amd64-aws-us-east-2_clusterpool.yaml [hunk 2]
```diff
@@ -0,0 +1,41 @@
+apiVersion: hive.openshift.io/v1
+kind: ClusterPool
+metadata:
+  creationTimestamp: null
+  labels:
+    architecture: amd64
+    cloud: aws
+    owner: konflux
+    product: ocp
+    region: us-east-2
```
Comment (flacatus) [inline clusters/hosted-mgmt/hive/pools/konflux/konflux-ocp-4-15-0-amd64-aws-us-east-2_clusterpool.yaml:10]: Is not us-east-1 more cheaper?

Comment (psturc) [reply clusters/hosted-mgmt/hive/pools/konflux/konflux-ocp-4-15-0-amd64-aws-us-east-2_clusterpool.yaml:10]: https://www.concurrencylabs.com/blog/choose-your-aws-region-wisely/
oregon (us-west-2), n. virginia (us-east-1), and ohio (us-east-2) are roughly the same regarding cost

we can use us-east-1 for hcp clusters

**Review by kasemAlem — APPROVED** [review]

**Pull request merged**